contrib/dracut/90zfs/zfs-generator.sh.in

```diff
@@ -15,8 +15,10 @@ type getarg >/dev/null 2>&1 || . "$dracutlib"
 [ "${root##zfs:}" = "${root}" -a "${root##ZFS=}" = "${root}" -a "$rootfstype" != "zfs" ] && exit 0
 
 rootfstype=zfs
-if echo "${rootflags}" | grep -q zfsutil ; then
+if echo "${rootflags}" | grep -Eq '^zfsutil$|^zfsutil,|,zfsutil$|,zfsutil,' ; then
     true
+elif test -n "${rootflags}" ; then
+    rootflags="zfsutil,${rootflags}"
 else
     rootflags=zfsutil
 fi
```

Comment:
Nothing about what I said has anything to do with adding fstab to your initrd.  Additionally, nothing in this changeset changes anything about the way that file systems are mounted *after* the system is done with the initial RAM disk.

The program I linked above is simply an enhancement that inspects /etc/fstab *at the time of the creation of grub.cfg*, and then adds the proper `root=ZFS/dataset` parameter to the kernel command line.  This only happens when the system is already fully booted and you (or something else) run `grub2-mkconfig`.